### PR TITLE
Test certmonger-generated Cockpit certificates

### PR DESCRIPTION
**Pull request by martinpitt**
Add an integration test that generates a certificate with certmonger,
through the LSR certificate role. For this to work on RHEL/CentOS 7 this
has to employ a run_after script, which merges the cert and key [1],
renames it to "*.cert" [2], and fixes the permissions [3].

For older distributions this requires a hack to make
/etc/cockpit/ws-certs.d/ writable for certmonger [4].

[1] Cockpit 210 introduced support for separate *.key files in addition
    to a merged cert/key file.
[2] Cockpit 210 looks for *.crt files in addition to *.cert
[3] lsr.certificates sets wrong permissions:
    https://github.com/linux-system-roles/certificate/pull/97
[4] Cockpit 255 fixes its SELinux policy by default, so Fedora and
    CentOS/RHEL ≥ 9.0 don't need this any more.

-----
This is ugly, and PR #37 shows how it's going to look like in the future (past RHEL 7), with the necessary fixes in Cockpit and LSR.

I would like to land something like this to keep ensuring that this approach works. After that, I would like to discuss whether we want to document this as the "official" solution, or want to add some syntactic sugar for it.

This PR avoids all the know problems above, but it *still* does not work for me in a RHEL 7.9 VM: While cockpit uses the self-signed certificate now, curl/OpenSSL don't accept it:
```
# openssl s_client -CAfile /etc/pki/tls/certs/monger-cockpit.crt -verify 1 -verify_return_error -connect localhost:9090 </dev/null 
verify depth is 1
CONNECTED(00000003)
depth=0 CN = localhost
verify error:num=20:unable to get local issuer certificate
139932372125584:error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed:s3_clnt.c:1264:
[...]
```

 This works fine in other OSes, let's see what CI says.

Also, I think @richm mentioned something about how to properly rope in `linux-system-roles.certificate` for the integration tests: how can they use the role from the RPM instead of checking out git?

 - [x] RHEL 7 OpenSSL does not accept self-signed certificate: https://github.com/linux-system-roles/certificate/issues/98 and https://github.com/linux-system-roles/certificate/issues/99 -- worked around
 - [x] Use lsr RPM for integration tests

### CODE REVIEW

**Comment by martinpitt** [comment]
Ooh, I found out why this does not work on RHEL 7: The request is for `ca: self-sign`, but it ends up as
```
Request ID '20211020094829':
	status: MONITORING
	stuck: no
	key pair storage: type=FILE,location='/etc/pki/tls/private/monger-cockpit.key'
	certificate: type=FILE,location='/etc/pki/tls/certs/monger-cockpit.crt'
	CA: local
```

The "local" CA is hidden *very* well in /var/lib/certmonger/cas/20211020094828-4 , and I did not find documentation or a public API how to actually get the CA certificate (`getcert list-cas -c local` doesn't tell at least). So this is a bug in the lsr.certificate role, that it does not properly implement `ca: self-sign` on RHEL.

`curl -vv --cacert /var/lib/certmonger/cas/20211020094828-4 https://localhost:9090` works fine, but of course this is neither stable, nor an API, not even supposed to be that way.

**Comment by richm** [comment]
[citest pending]

File: tests/tests_certificate.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+- name: test - setup
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: Download current linux-system-roles.certificate
+      git:
+        repo: https://github.com/linux-system-roles/certificate/
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply tests/tests_certificate.yml:17]: Hmm, it seems `$BEAKERLIB` is not defined on [centos-7](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-39-5da7fbb-centos-7-20211021-114624/artifacts/ansible.log), [rhel-8](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-39-5da7fbb-rhel-8-y-20211021-114706/artifacts/ansible.log) nor all the other integration tests. `__is_beaker_env` is false on all of them.

[the remark this answers is not in the document]
Comment (martinpitt) [reply tests/tests_certificate.yml:8]: Ack, done.

**Pull request merged**